lib/utils.js

```diff
@@ -87,6 +87,8 @@ exports.GreatResponse = function (request, options, filterRules) {
 
     var applyFilter = function (data) {
 
+        if(!data || Object.keys(data).length === 0) return;
```

Comment:
I think the check should be for `!(data && typeof data === 'object')`. While `null` or `undefined` cause an issue, plain string responses or payloads cause the problem too.